dvc/fs/local.py

```diff
@@ -94,18 +64,14 @@ def walk(
         for root, dirs, files in os.walk(
             top, topdown=topdown, onerror=onerror
         ):
-            if use_dvcignore:
-                dirs[:], files[:] = self.dvcignore(
-                    os.path.abspath(root),
-                    dirs,
-                    files,
-                    ignore_subrepos=ignore_subrepos,
-                )
-
+            dirs[:] = list(filter(lambda i: i not in self.HIDDEN_DIRS, dirs))
             yield os.path.normpath(root), dirs, files
 
     def walk_files(self, path_info, **kwargs):
-        for root, _, files in self.walk(path_info):
+        ignore_filter = kwargs.get("ignore_filter", None)
+        for root, dirs, files in self.walk(path_info):
+            if ignore_filter:
+                dirs, files = ignore_filter(root, dirs, files)
```

Comment:
How about we just call `ignore_filter` outside, when we need it? E.g. by wrapping walk_files result with it. That way it won't leak into the fs level.